src/combinatorics/set_family/compression/uv.lean

```diff
@@ -0,0 +1,265 @@
+/-
+Copyright (c) 2021 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import data.finset.basic
+
+/-!
+# UV-compressions
+
+This file defines UV-compression.
+
+A compression in combinatorics is an operation on a set family that reduces its shadow.
```

Comment:
This phrasing suggests that compressions are only used in this context of set families, but more general operations are also called compressions (see e.g. 'Compressions and Isoperimetric Inequalities' by Bollobás and Leader).

Reply:
How would you describe compressions in general? This paper from Bollobás and Leader is the only one I could find that mentions compression in the context of combinatorics.

Reply:
It's an umbrella term for a kind of technique. In general you have some structure V, some notion of "size" and "perimeter" of subsets of V, and define compression operators on the subsets of V which preserve size and reduce perimeter. But I don't think you need to discuss the general case, something along the lines of "UV-compression is an operation on a set family that reduces its shadow" would be fine.